File: rb/features.rb
# RandomUserGenerator SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module RandomUserGeneratorFeatures
  def self.make_feature(name)
    case name
    when "base"
      RandomUserGeneratorBaseFeature.new
    when "test"
      RandomUserGeneratorTestFeature.new
    else
      RandomUserGeneratorBaseFeature.new
    end
  end
end

File: py/randomusergenerator_sdk/features.py
# RandomUserGenerator SDK feature factory

from randomusergenerator_sdk.feature.base_feature import RandomUserGeneratorBaseFeature
from randomusergenerator_sdk.feature.test_feature import RandomUserGeneratorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RandomUserGeneratorBaseFeature(),
        "test": lambda: RandomUserGeneratorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
